src/components/external-events/ExternalEventForm.svelte

```diff
@@ -0,0 +1,136 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import { field } from '../../stores/form';
+  import { plugins } from '../../stores/plugins';
+  import type { ExternalEvent } from '../../types/external-event';
+  import type { ExternalEventProperty } from '../../types/external-event-property';
+  import type { FieldStore } from '../../types/form';
+  import { formatDate } from '../../utilities/time';
+  import Collapse from '../Collapse.svelte';
+  import DatePickerField from '../form/DatePickerField.svelte';
+  import Input from '../form/Input.svelte';
+  import Properties from './Properties.svelte';
+
+  export let externalEvent: ExternalEvent;
+  export let showHeader: boolean = true;
+
+  let startTimeField: FieldStore<string>;
+  let formProperties: ExternalEventProperty[] = [];
+
+  $: startTimeField = field<string>(`${formatDate(new Date(externalEvent.start_time), $plugins.time.primary.format)}`);
+  $: formProperties = Object.entries(externalEvent.properties).map(externalEvent => {
+    return {
+      name: externalEvent[0],
+      value: externalEvent[1],
+    };
+  });
+</script>
+
+<div class="external-event-form-container">
+  {#if showHeader}
+    <div class="external-event-header">
+      <div class="external-event-header-title">
+        <div class="external-event-header-title-value st-typography-medium">
+          {externalEvent.pkey.key}
+        </div>
+      </div>
+    </div>
+  {/if}
+
+  <div class="external-event-form">
+    <fieldset>
+      <Collapse title="Definition">
+        <Input layout="inline">
+          Type
+          <input class="st-input w-100" disabled={true} name="event-type" value={externalEvent.pkey.event_type_name} />
+        </Input>
+
+        <DatePickerField
+          layout="inline"
+          disabled={true}
+          field={startTimeField}
+          label={`Start Time (${$plugins.time.primary.label})`}
+          name="start-time"
+        />
+
+        <Input layout="inline">
+          Duration
+          <input class="st-input w-100" disabled={true} name="duration" value={externalEvent.duration} />
+        </Input>
+
+        <Input layout="inline">
+          Source ID
+          <input class="st-input w-100" disabled={true} name="source-key" value={externalEvent.pkey.source_key} />
+        </Input>
+
+        <Input layout="inline">
+          Source File
+          <input
+            class="st-input w-100"
+            disabled={true}
+            name="source-key-resolved"
+            value={externalEvent.pkey.source_key}
+          />
+        </Input>
+      </Collapse>
+    </fieldset>
+
+    <fieldset>
+      <Collapse title={formProperties.length > 0 ? `Properties` : ''}>
```

Comment:
Can we not remove the title if no properties are found? Looks odd to do this. Could just always say properties, the message underneath should suffice. We use this pattern elsewhere so should be familiar to users.
<img width="330" alt="image" src="https://github.com/user-attachments/assets/3e42fbea-1629-410c-ba62-492d9cb9f59a">